dask_ml/model_selection/_incremental.py

```diff
@@ -347,3 +381,309 @@ def fit(
+    """Base class for estimators using the incremental `fit`.
+
+    Subclasses must implement the following abstract method
+
+    * _additional_calls
+    """
+
+    def __init__(
+        self,
+        estimator,
+        parameters,
+        test_size=0.15,
+        random_state=None,
+        scoring=None,
+        iid=True,
+        refit=True,
+        error_score="raise",
+        return_train_score=_RETURN_TRAIN_SCORE_DEFAULT,
+        scheduler=None,
+        n_jobs=-1,
+        cache_cv=True,
+    ):
+        # TODO: find the subset of sensible parameters.
+        self.parameters = parameters
+        self.test_size = test_size
+        self.random_state = random_state
+        super(BaseIncrementalSearch, self).__init__(
+            estimator,
+            scoring=scoring,
+            iid=iid,
+            refit=refit,
+            error_score=error_score,
+            return_train_score=return_train_score,
+            scheduler=scheduler,
+            n_jobs=n_jobs,
+            cache_cv=cache_cv,
+        )
+
+    def _check_array(self, X, y, **kwargs):
+        """Validate the data arguments X and y.
+
+        By default, NumPy arrays are converted to 1-block dask arrays.
+
+        Parameters
+        ----------
+        X, y : array-like
+        """
+        if isinstance(X, np.ndarray):
+            X = da.from_array(X, X.shape)
+        if isinstance(y, np.ndarray):
+            y = da.from_array(y, y.shape)
+        X = check_array(X, **kwargs)
+        kwargs["ensure_2d"] = False
+        y = check_array(y, **kwargs)
+        return X, y
+
+    def _get_train_test_split(self, X, y, **kwargs):
+        """CV-Split the arrays X and y
+
+        By default, :meth:`dask_ml.model_selection.train_test_split`
+        is used with ``self.test_size``. The test set is expected to
+        fit in memory on each worker machine.
+
+        Parameters
+        ----------
+        X, y : dask.array.Array
+        """
+        X_train, X_test, y_train, y_test = train_test_split(
+            X, y, test_size=self.test_size
+        )
+        return X_train, X_test, y_train, y_test
+
+    def _additional_calls(self):
+        raise NotImplementedError
+
+    def _get_params(self):
+        """Parameters to pass to `fit`.
+
+        By defualt, a GridSearch over ``self.parameters`` is used.
+        """
+        return ParameterGrid(self.parameters)
+
+    def _get_history_results(self, results):
+        # type: (Results) -> Dict
+        """Construct the CV results.
+
+        Has the following keys:
+
+        * params
+        * test_score
+        * mean_test_score
+        * rank_test_score
+        * mean_partial_fit_time
+        * std_partial_fit_time
+        * mean_score_time
+        * std_score_time
+        * partial_fit_calls
+        * model_id
+        """
+        info, model, history = results
+        key = operator.itemgetter("model_id")
+        hist2 = sorted(history, key=key)
```

Comment:
Can you try sorting by the tuple `(model_id, partial_fit_calls)` here? That may fix the history issue.

Reply:
To check my understanding, is it true that for a given `model_id`, `partial_fit_calls` is always increasing as we move through history?

Reply:
Yes